sysfs/net_class.go

```diff
@@ -0,0 +1,135 @@
+	MTU              uint64 `fileName:"mtu"`                // /sys/class/net/<iface>/mtu
+	NameAssignType   uint64 `fileName:"name_assign_type"`   // /sys/class/net/<iface>/name_assign_type
+	NetDevGroup      uint64 `fileName:"netdev_group"`       // /sys/class/net/<iface>/netdev_group
+	OperState        string `fileName:"operstate"`          // /sys/class/net/<iface>/operstate
+	PhysPortID       string `fileName:"phys_port_id"`       // /sys/class/net/<iface>/phys_port_id
+	PhysPortName     string `fileName:"phys_port_name"`     // /sys/class/net/<iface>/phys_port_name
+	PhysSwitchID     string `fileName:"phys_switch_id"`     // /sys/class/net/<iface>/phys_switch_id
+	Speed            uint64 `fileName:"speed"`              // /sys/class/net/<iface>/speed
+	TxQueueLen       uint64 `fileName:"tx_queue_len"`       // /sys/class/net/<iface>/tx_queue_len
+	Type             uint64 `fileName:"type"`               // /sys/class/net/<iface>/type
+}
+
+// NetClass is collection of info for every interface in /sys/class/net. The map keys
+// are interface names.
+type NetClass map[string]NetClassInterface
+
+// NewNetClass returns info for all net interfaces read from /sys/class/net/<interface>.
+func NewNetClass() (NetClass, error) {
+	fs, err := NewFS(DefaultMountPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return fs.NewNetClass()
+}
+
+// NewNetClass returns info for all net interfaces read from /sys/class/net/<interface>.
+func (fs FS) NewNetClass() (NetClass, error) {
+	return newNetClass(fs.Path("class/net"))
+}
+
+// NewNetClass returns info for all net interfaces read from /sys/class/net/<interface>.
+func newNetClass(path string) (NetClass, error) {
+	devices, err := ioutil.ReadDir(path)
+	if err != nil {
+		return NetClass{}, fmt.Errorf("cannot access %s dir %s", path, err)
+	}
+
+	netClass := NetClass{}
+	for _, deviceDir := range devices {
+		interfaceClass, err := netClass.parseInterfaceNetClass(path + "/" + deviceDir.Name())
+		if err != nil {
+			continue
+		}
+		interfaceClass.Name = deviceDir.Name()
+		netClass[deviceDir.Name()] = *interfaceClass
+	}
+	return netClass, nil
+}
+
+// parseInterfaceNetClass scans predefined files in /sys/class/net/<interface>
+// directory and gets their contents.
+func (nc NetClass) parseInterfaceNetClass(devicePath string) (*NetClassInterface, error) {
+	interfaceClass := NetClassInterface{}
+	interfaceElem := reflect.ValueOf(&interfaceClass).Elem()
+	interfaceType := reflect.TypeOf(interfaceClass)
+
+	for i := 0; i < interfaceElem.NumField(); i++ {
+		fieldType := interfaceType.Field(i)
+		fieldValue := interfaceElem.Field(i)
+
+		if fieldType.Tag.Get("fileName") == "" {
+			continue
+		}
+
+		fileContents, err := ioutil.ReadFile(devicePath + "/" + fieldType.Tag.Get("fileName"))
+		if err != nil {
+			continue
+		}
+		value := strings.TrimSpace(string(fileContents))
+
+		if fieldValue.Kind() == reflect.Uint64 {
```

Comment:
I'd suggest to use a switch statement here and error handling in the default case, to ensure all value types are handled.

```go
switch fieldValue.Kind() {
case reflect.Uint64:
// ...
default:
  return nil, fmt.Errorf("unhandled type %q", fieldValue.Kind())
}